Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -2058,6 +2058,28 @@ int PhysicalParticleContainer::GetRefineFac(const Real x, const Real y, const Re
     return ref_fac;
 }
 
+/*Function to store values for back-tranformed diagnostic"*/
+void
+PhysicalParticleContainer::warpx_copy_attribs(const long* np,
+								const amrex::Real* xp, const amrex::Real* yp, const amrex::Real* zp,
+								const amrex::Real* uxp, const amrex::Real* uyp, const amrex::Real* uzp,
+								amrex::Real* xpold, amrex::Real* ypold, amrex::Real* zpold,
+								amrex::Real* uxpold, amrex::Real* uypold, amrex::Real* uzpold)
+{
+
+	for(int n=0; n<*np; ++n)
```

Comment:
This loop is perfectly working on CPU, but would not be suitable for GPU. To handle both CPU and GPU programming, we use an abstract layer provided by AMReX: we call a function `ParallelFor` that does different things, depending on whether you compile for CPU or GPU. A basic example can be found at `PhysicalParticleContainer::ConvertUnits`. In this case, I think you could do something like:
```
ParallelFor( np,
    [=] AMREX_GPU_DEVICE (long i) {
        xpold[n]=xp[n];
        ypold[n]=yp[n];
        zpold[n]=zp[n];
        uxpold[n]=uxp[n];
        uypold[n]=uyp[n];
        uzpold[n]=uzp[n];
    }
);
```
Thank we can discuss a general way to get all these arrays.